### PR TITLE
3DS Security Update

### PR DESCRIPTION
**Pull request by pritesh-lr**

### CODE REVIEW

**Comment by pritesh-lr** [comment]
The key and xkey are encrypted and don't display the original keys.
image

After testing all functionality include with 3Ds functionality.
image

**Comment by pritesh-lr** [comment]
Fixed the issue related **xKey**

image

image

image

check using this test card: **5200 0000 0000 2151** - 3DS Card.

File: includes/class-wc-gateway-cardknox.php [hunk 1]
```diff
@@ -419,9 +418,13 @@ public function payment_scripts()
 
         wp_enqueue_script('cardknox', 'https://cdn.cardknox.com/ifields/2.15.2401.3101/ifields.min.js', '', '1.0.0', false);
         wp_enqueue_script('woocommerce_cardknox', plugins_url('assets/js/cardknox' . $suffix . '.js', WC_CARDKNOX_MAIN_FILE), array('jquery-payment'), filemtime(plugin_dir_path(dirname(__FILE__)) . 'assets/js/cardknox' . $suffix . '.js'), true);
+        
+
+        $encrypted_key = $this->token_key;
+        //$encrypted_xkey = $this->transaction_key;
```
Comment (mstein-ck) [inline includes/class-wc-gateway-cardknox.php:424]: Remove comments?

Comment (pritesh-lr) [reply includes/class-wc-gateway-cardknox.php:424]: Hello @mstein-ck 
Remove Commented Code
image

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## changelog for version Version: 1.2.71
+
+- 3DS Key is exposed issue Fixed.
```
Comment (mstein-ck) [inline changelog.md:5]:
```suggestion
- Security Updates
```

File: includes/class-wc-gateway-cardknox.php [hunk 3]
```diff
@@ -419,9 +418,12 @@ public function payment_scripts()
 
         wp_enqueue_script('cardknox', 'https://cdn.cardknox.com/ifields/2.15.2401.3101/ifields.min.js', '', '1.0.0', false);
         wp_enqueue_script('woocommerce_cardknox', plugins_url('assets/js/cardknox' . $suffix . '.js', WC_CARDKNOX_MAIN_FILE), array('jquery-payment'), filemtime(plugin_dir_path(dirname(__FILE__)) . 'assets/js/cardknox' . $suffix . '.js'), true);
+        
+
+        $encrypted_key = $this->token_key;
```
Comment (mstein-ck) [inline includes/class-wc-gateway-cardknox.php:423]: Is this key encrypted? Who is decrypting it?

Comment (pritesh-lr) [reply includes/class-wc-gateway-cardknox.php:423]: Hello @mstein-ck 
No, this key is not encrypted.it's just a variable name.

Comment (pritesh-lr) [reply includes/class-wc-gateway-cardknox.php]: @mstein-ck Changed the variable name.

image

**Review by mstein-ck — APPROVED** [review]

**Pull request merged**